packages/devextreme/js/__internal/scheduler/tooltip_strategies/m_tooltip_strategy_base.ts

```diff
@@ -139,15 +141,25 @@ export class TooltipStrategyBase {
 
     const isEmptyDropDownAppointmentTemplate = this._isEmptyDropDownAppointmentTemplate();
     // @ts-expect-error
-    return new FunctionTemplate((options) => template.render({
-      model: isEmptyDropDownAppointmentTemplate ? {
-        appointmentData,
-        targetedAppointmentData,
-        isButtonClicked,
-      } : appointmentData,
-      container: options.container,
-      index,
-    }));
+    return new FunctionTemplate((options) => {
+      // eslint-disable-next-line @typescript-eslint/init-declarations
+      let resolve: () => void;
+      const promise = new Promise<void>((r) => { resolve = r; });
```

Comment:
Question.

Can we use the ```deferred``` here?
Or if we want to remove dependency from our custom ```deferred```, let's create new ```deferred-like``` helper object in ```__internal/core``` :)

Reply:
Why?
If you're worried about this ugly `new Promise(r => resolve = r)`, it will be changed after we'll have access to `Promise.withResolvers` API (we can't because our TS is old) https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers

Reply:
Yes, I worried about this ugly ```new Promise(r => resolve = r)``` and that this custom construction to manage the promise.
In this place, you implement it this way, in other places other developers can implement it another way, and so on.
It's better to use one approach across the codebase.

I see two solutions here:

1. Use our ```deferred``` and replace it sometime in the future as a separate refactoring task
(now it's not in our plans, so in other words just use ```deferred``` for now)
Because we now use it everywhere :)
2. Create ```newDeferred``` object somewhere in the ```__internal/core``` and use it here.
I think it will be better to do it in a separate PR.
And start slowly migrate from our old ```deferred``` to ```newDeferred``` :)

Reply:
newDeferred is no better than deferred
old Deferred is bad
This ugly construction is well-known is used in many projects
Moving to new API (.withResolvers) will be easier if we have Promise than if we would have Deferred (because in the future we'll change only this part of code, keeping previous typing)

Reply:
Ok, got you
Maybe we should add the next utils function in the future to make this less ugly?
Something like this (and maybe we should add a "reject" too):
```typescript
export const createHandledPromise= <T>() => {
  let resolve: (value: T) => void;
  const promise = new Promise<T>((r) => resolve);
  return [resolve, promise];
}
```